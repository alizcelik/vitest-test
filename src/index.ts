export function deepMerge(a, b) {
  if (Array.isArray(a) && Array.isArray(b)) {
    return [...a, ...b];
  }

  return Object.assign(a, b);
}
