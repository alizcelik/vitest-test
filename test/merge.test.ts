import { test, expect } from "vitest";
import { deepMerge } from "../src";

test("shallow merge", () => {
  const merged = deepMerge(
    {
      name: "Ali",
    },
    {
      github: "aliozclk",
    }
  );

  expect(merged).toEqual({
    name: "Ali",
    github: "aliozclk",
  });
});

test("shallow merge with overlaps", () => {
  const merged = deepMerge(
    {
      name: "Ali",
      github: "unknown",
    },
    {
      github: "aliozclk",
      twitter: "aliozclk99",
    }
  );

  expect(merged).toMatchSnapshot();
});

test("shallow merge with arrays", () => {
  const merged = deepMerge(["vue", "react"], ["svelte", "angular"]);

  expect(merged).toEqual(["vue", "react", "svelte", "angular"]);
});

test("deep merge with overlaps", () => {
  const merged = deepMerge(
    {
      name: "Ali",
      accounts: {
        github: "unknown",
      },
      languages: ["javascript"],
    },
    {
      accounts: {
        github: "aliozclk",
        twitter: "aliozclk99",
      },
      languages: ["typescript", "vue"],
    }
  );

  expect(merged).toMatchInlineSnapshot(`
    {
      "accounts": {
        "github": "aliozclk",
        "twitter": "aliozclk99",
      },
      "languages": [
        "javascript",
        "typescript",
        "vue",
      ],
      "name": "Ali",
    }
  `);
});

test("throws error when merging arrays with objects", () => {
  expect(() => deepMerge(["foo", "bar"], { foo: "bar" })).toThrowError(
    "Can not merge two differnet types"
  );
});
