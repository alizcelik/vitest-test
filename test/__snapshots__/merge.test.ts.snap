// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shallow merge with overlaps 1`] = `
{
  "github": "aliozclk",
  "name": "Ali",
  "twitter": "aliozclk99",
}
`;
